streams/src/main/java/org/apache/kafka/streams/processor/internals/namedtopology/KafkaStreamsNamedTopologyWrapper.java

```diff
@@ -239,12 +238,16 @@ public RemoveNamedTopologyResult removeNamedTopology(final String topologyToRemo
         final boolean skipResetForUnstartedApplication =
             maybeCompleteFutureIfStillInCREATED(removeTopologyFuture, "removing topology " + topologyToRemove);
 
-        if (resetOffsets && !skipResetForUnstartedApplication) {
+        if (resetOffsets && !skipResetForUnstartedApplication && !partitionsToReset.isEmpty()) {
```

Comment:
Moved the `!partitionsToReset.isEmpty()` check here to make sure we don't log the line about resetting offsets if we don't actually have any offsets to reset